### PR TITLE
Add a template for new Pull Requests

### PR DESCRIPTION
**Pull request by galanCA**
Adds a template for pull requests. This will provide a guide on the steps that the author needs to take to open and merge a pull request.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - [ ] I have only committed the necessary changes for this fix or feature
+    - [ ] I have made corresponding changes to the documentation
+    - [ ] My changes generate no new warnings
+    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
+        - [ ] exercising the code changes in the test framework, and/or
+        - [ ] manually verifying the changes (as explained in the the pull request description above)
+    - [ ] My changes pass all local tests
+    - [ ] My changes successfully passes CI checks
+    - [ ] Add any unit test for proof and documentation.
+- [ ] Move pull request out of draft mode and assign reviewers
```
Comment (nealkruis) [inline .github/pull_request_template.md:28]: We should add "Merge in main branch and address resulting conflicts and/or test failures." before it is pulled out of draft mode and reviewers are assigned.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -25,10 +25,12 @@ Replace the content in this section with:
         - [ ] manually verifying the changes (as explained in the the pull request description above)
     - [ ] My changes pass all local tests
     - [ ] My changes successfully passes CI checks
+    - [ ] Add any unit test for proof and documentation.
 - [ ] Move pull request out of draft mode and assign reviewers
 - [ ] Iterate with reviewers until all changes are approved
     - [ ] Make changes in response to reviewer comments
     - [ ] Re-request review in GitHub
+    - [ ] Add Chip for final review.
```
Comment (nealkruis) [inline .github/pull_request_template.md:32]: This isn't necessary here. It is more of an informal rule.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ - [ ] Confirm all CI checks pass and there are no build warnings
+ - [ ] Pull, build, and run automated tests locally
+ - [ ] Perform manual tests of the fix or feature locally
+ - [ ] Add any review comments, if applicable
+ - [ ] Submit review in GitHub as either
+     - [ ] Request changes, or
+     - [ ] Approve
+ - [ ] Iterate with author until all changes are approved
+
+ ## Final Check
```
Comment (nealkruis) [inline .github/pull_request_template.md:47]: I don't think the "Final check" is necessary if we move the merge of the main branch as recommended.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
+        - [ ] exercising the code changes in the test framework, and/or
+        - [ ] manually verifying the changes (as explained in the the pull request description above)
+    - [ ] My changes pass all local tests
+    - [ ] My changes successfully passes CI checks
+    - [ ] Add any unit test for proof and documentation.
+- [ ] Move pull request out of draft mode and assign reviewers
+- [ ] Iterate with reviewers until all changes are approved
+    - [ ] Make changes in response to reviewer comments
+    - [ ] Re-request review in GitHub
```
Comment (nealkruis) [inline .github/pull_request_template.md:31]: Add it again before "Re-request review in GitHub."

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 4 above)
Comment (galanCA) [reply .github/pull_request_template.md:33]: Should we add a check to re-review warnings?

Comment (nealkruis) [reply .github/pull_request_template.md:33]: Ideally, any changes made during the iteration process should also include the full self-review checklist.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+- A summary of the behavior expected from this change
+- A description of tests performed
+
+## Author Progress Checklist:
+
+- [ ] Open draft pull request
+    - [ ] Make title clearly understandable in a standalone change log context
+    - [ ] Assign yourself the issue
+    - [ ] Add at least one label
+- [ ] Make code changes (if you haven't already)
```
Comment (nealkruis) [inline .github/pull_request_template.md:15]: We should add linking to the issue(s) in the Development field of the PR:

Screen Shot 2022-04-01 at 10 41 44 AM

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -11,7 +11,8 @@ Replace the content in this section with:
 - [ ] Open draft pull request
     - [ ] Make title clearly understandable in a standalone change log context
     - [ ] Assign yourself the issue
-    - [ ] Add at least one label
+    - [ ] Add at least one label (enhancement, bug, and/or maintenance)
+    - [ ] Link the issue(s) addressed by this PR (under the "Development" sidebar)
```
Comment (nealkruis) [inline .github/pull_request_template.md:15]: > under "Development" in the sidebar menu

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -11,7 +11,8 @@ Replace the content in this section with:
 - [ ] Open draft pull request
     - [ ] Make title clearly understandable in a standalone change log context
     - [ ] Assign yourself the issue
-    - [ ] Add at least one label
+    - [ ] Add at least one label (enhancement, bug, and/or maintenance)
```
Comment (nealkruis) [inline .github/pull_request_template.md:14]: These examples are mutually exclusive.

**Review by nealkruis — CHANGES_REQUESTED** [review]

**Review by nealkruis — APPROVED** [review]

**Pull request merged**